src/annotator/integrations/pdf2htmlEx.tsx

```diff
@@ -0,0 +1,551 @@
+      }
+    }
+
+    // Find all the anchors that have been invalidated by page state changes.
+    for (const anchor of this._annotator.anchors) {
+      // Skip any we already know about.
+      if (anchor.highlights) {
+        if (refreshAnnotations.includes(anchor.annotation)) {
+          continue;
+        }
+
+        // If the highlights are no longer in the document it means that either
+        // the page was destroyed by PDF.js or the placeholder was removed above.
+        // The annotations for these anchors need to be refreshed.
+        for (let index = 0; index < anchor.highlights.length; index++) {
+          const hl = anchor.highlights[index];
+
+          if (!document.body.contains(hl)) {
+            anchor.highlights.splice(index, 1);
+            delete anchor.range;
+            refreshAnnotations.push(anchor.annotation);
+            break;
+          }
+        }
+      }
+    }
+
+    refreshAnnotations.map(annotation => this._annotator.anchor(annotation));
+  }
+
+  /**
+   * Return the scrollable element which contains the document content.
+   */
+  contentContainer(): HTMLElement {
+    return this._pdfViewer.container;
+  }
+
+  /**
+   * Attempt to make the PDF viewer and the sidebar fit side-by-side without
+   * overlap if there is enough room in the viewport to do so reasonably.
+   * Resize the PDF viewer container element to leave the right amount of room
+   * for the sidebar, and prompt PDF.js to re-render the PDF pages to scale
+   * within that resized container.
+   *
+   * @return - True if side-by-side mode was activated
+   */
+  fitSideBySide(sidebarLayout: SidebarLayout): boolean {
+    const maximumWidthToFit = window.innerWidth - sidebarLayout.width;
+    const active = sidebarLayout.expanded && maximumWidthToFit >= MIN_PDF_WIDTH;
+
+    // If the sidebar is closed, we reserve enough space for the toolbar controls
+    // so that they don't overlap a) the chevron-menu on the right side of
+    // PDF.js's top toolbar and b) the document's scrollbar.
+    //
+    // If the sidebar is open, we reserve space for the whole sidebar if there is
+    // room, otherwise we reserve the same space as in the closed state to
+    // prevent the PDF content shifting when opening and closing the sidebar.
+    const reservedSpace = active
+      ? sidebarLayout.width
+      : sidebarLayout.toolbarWidth;
+    this._pdfContainer.style.width = `calc(100% - ${reservedSpace}px)`;
+
+    // The following logic is pulled from PDF.js `webViewerResize`
+    // const currentScaleValue = this._pdfViewer.currentScaleValue;
+    // if (
+    //   currentScaleValue === 'auto' ||
+    //   currentScaleValue === 'page-fit' ||
+    //   currentScaleValue === 'page-width'
+    // ) {
+    //   // NB: There is logic within the setter for `currentScaleValue`
+    //   // Setting this scale value will prompt PDF.js to recalculate viewport
+    //   this._pdfViewer.currentScaleValue = currentScaleValue;
```

Comment:
I didn't test side by side yet, we may have to add some styles to the pdf2HtmlEx container.